### PR TITLE
UPSTREAM: 48936: AWS: check validity of KSM key before creating a new encrypted disk

### PR DESCRIPTION
**Pull request by jsafrane**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1467244

cc @openshift/sig-storage

### CODE REVIEW

**Comment by jsafrane** [comment]
I rearranged changes in godeps and vendor/AWS to pass `hack/verify-upstream-commits.sh`, I am not sure I got it right.

Goal of this exercise is to get https://github.com/kubernetes/kubernetes/pull/48936/files without any code changes, just with the right patch structure so next Kubernetes rebase does not conflict much.

**Comment by jsafrane** [comment]
@deads2k @liggit, PTAL at the commit messages so next rebase is flawless
/assign @hekumar @rootfs for full code review

File: Godeps/Godeps.json [hunk 1]
```diff
@@ -370,6 +370,11 @@
 			"Rev": "63ce630574a5ec05ecd8e8de5cea16332a5a684d"
 		},
 		{
+			"ImportPath": "github.com/aws/aws-sdk-go/service/kms",
```
Comment (deads2k) [inline Godeps/Godeps.json]: This should be in the same commit that has the code.  Shaped like `bump(github.com/aws/aws-sdk-go): 63ce630574a5ec05ecd8e8de5cea16332a5a684d`.  Also, thanks for picking the same level as everything else.

File: Godeps/Godeps.json [hunk 2]
```diff
@@ -370,6 +370,11 @@
...
+			"Comment": "v1.6.10",
+			"Rev": "63ce630574a5ec05ecd8e8de5cea16332a5a684d"
```
Comment (deads2k) [inline Godeps/Godeps.json]: is upstream on this level too?

Comment (jsafrane) [reply Godeps/Godeps.json]: yes, current kubernetes master is on v1.6.10

File: Godeps/Godeps.json (hunk 1 above)
Comment (jsafrane) [reply Godeps/Godeps.json]: Fixed, thanks

**Comment by jsafrane** [comment]
Can we merge this? It fixes a bug and AFAIK today is the deadline.

**Comment by jsafrane** [comment]
Guys, this PR is open for 2 months and has already slipped 3.7. Can we get it into 3.8?

**Comment by deads2k** [comment]
> Guys, this PR is open for 2 months and has already slipped 3.7. Can we get it into 3.8?

Didn't we pick it up in the rebase already?

**Comment by liggitt** [comment]
yes, it's already in master

**Pull request closed without merging**